modules/bibauthorid/lib/bibauthorid_hoover.py

```diff
@@ -0,0 +1,715 @@
+            tmp = dict_entry_for_hepnames_connector(pid, inspireid)
+            if tmp:
+                self.cname_dict.update(tmp)
+            if len(self.cname_dict.keys()) >= self.packet_size:
+                self.execute_connection()
+
+    def _execute_connection(self):
+        """Execute the connection between the canonical name and the hepnames
+        record. In case the dry_hepnames_run option is true this functions is
+        substituted with the _null_func(), a function that has no effect.
+        """
+        add_cname_to_hepname_record(self.cname_dict)
+        self.cname_dict.clear()
+
+
+def dict_entry_for_hepnames_connector(pid, inspireid):
+    """Produce the correct dictonary entry for the connector to use.
+
+    Arguments:
+    pid -- the pid of the author that has the inspireID
+    inspireID -- the inspireID of the author
+    """
+    author_canonical_name = get_canonical_name_of_author(pid)
+
+    if not author_canonical_name:
+        return None
+
+    recid = perform_request_search(p="035:" + inspireid, cc="HepNames")
+    if recid:
+        if len(recid) > 1:
+            raise MultipleHepnamesRecordsWithSameIdException(
+                "More than one hepnames record found with the same inspire id",
+                recid,
+                'INSPIREID')
+        write_message("Connecting pid %d canonical_name %s inspireID %s", (pid, author_canonical_name, inspireid), verbose=3)
+        return {author_canonical_name[0][0]: recid[0]}
+
+
+class Vacuumer(object):
+
+    """Class responsible for vacuuming the signatures to the right profile
+
+    Constructor arguments:
+    pid -- the pid of the author
+    """
+
+    def __init__(self, pid):
+        self.claimed_paper_signatures = set(
+            sig[1:4] for sig in get_papers_of_author(pid, include_unclaimed=False))
+        self.unclaimed_paper_signatures = set(
+            sig[1:4] for sig in get_papers_of_author(pid, include_claimed=False))
+        self.claimed_paper_records = set(
+            rec[2] for rec in self.claimed_paper_signatures)
+        self.unclaimed_paper_records = set(
+            rec[2] for rec in self.unclaimed_paper_signatures)
+        self.pid = pid
+
+    def vacuum_signature(self, signature):
+        if signature not in self.unclaimed_paper_signatures and signature not in self.claimed_paper_signatures:
+            if signature[2] in self.claimed_paper_records:
+                raise DuplicateClaimedPaperException(
+                    "Vacuum a duplicated claimed paper",
+                    self.pid,
+                    signature,
+                    filter(
+                        lambda x: x[2] == signature[2],
+                        self.claimed_paper_signatures))
+
+            duplicated_signatures = filter(
+                lambda x: signature[2] == x[2],
+                self.unclaimed_paper_signatures)
+
+            if duplicated_signatures:
+                write_message(("Conflict in pid %d",
```

Comment:
@petros-ioannidis that is the code that is currently failing in my test. BTW can you rebase meanwhile again against prod?